src/rgw/driver/d4n/rgw_sal_d4n.cc

```diff
@@ -255,28 +255,29 @@ int D4NFilterObject::get_obj_attrs(optional_yield y, const DoutPrefixProvider* d
   } else {
     /* Set metadata locally */
     RGWQuotaInfo quota_info;
-    RGWObjState* astate;
-    this->get_obj_state(dpp, &astate, y);
+    this->load_obj_state(dpp, y);
```

Comment:
this is the read-from-cache case, so i don't think we actually want to load anything from the backend here. can we call the setters below without this part?

Reply:
I admit, I don't know.  @samarahu do you have an opinion here?  Do we need to load info from the backend or is it already in the object?

Reply:
This is not needed in case of the read cache, because the attrs will anyway be fetched from the backend store (since the head/attrs are not being cached) and I believe get_obj_attrs() of the store anyway invokes get_obj_state internally.

But we need to get the object state in case of write-back cache where we are caching the head/attrs. We get the object state, and store some state variables as part of the cached head as attrs (like object_size, mtime, epoch etc), and those attrs will be read back while reading the head in case of a `get` and the the object state is set using set_obj_state. The PR is is still in progress.

Reply:
I can see that related getters/setters have been provided - they can be used in the PR that is in progress.

Reply:
This method isn't integral to any of the upstreamed D4N functionality. I just added this code as a temporary implementation for handling attributes within D4N. As Pritha mentioned, she is working on this part of the unified single-node cache for the next PR and based on her most recent response, it seems we can remove the call to the `get_obj_state` and replace it with specific getters and setters for the head metadata.

Reply:
Okay, I'll leave this as is, and let you guys work it out in future work.